handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -859,7 +859,21 @@ public final boolean isServer() {
      * Creates a new {@link SslHandler}.
      * <p>If {@link SslProvider#OPENSSL_REFCNT} is used then the returned {@link SslHandler} will release the engine
      * that is wrapped. If the returned {@link SslHandler} is not inserted into a pipeline then you may leak native
-     * memory!
+     * memory!</p>
+     * <p><b>Beware</b>: underlying generated {@link SSLEngine} won't have hostname verification enabled!
+     * If you create {@link SslHandler} for the client side and want proper security, we advice that you don't use
+     * this method. Instead, create a {@link SslHandler} manually with a properly configured {@link SSLEngine}:</p>
```

Comment:
I think I would just say that you need to enable it on the `SSLEngine`. This is also possible when you use `newHandler(....)`. 

```java
SslHandler handler = ...;
hander.engine().setSSLParameters(...)
```